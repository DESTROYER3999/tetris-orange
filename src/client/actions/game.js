import socket from "../services/socket-api";
import {
  KEY_PRESSED,
  UPDATE_SPECTRES,
  UPDATE_GAME_MODE
} from "../../constants/actionTypes";

export const handleKeyPress = e => {
  socket.emit(KEY_PRESSED, { code: e.code });
};

export const updateSpectres = spectres => ({
  type: UPDATE_SPECTRES,
  spectres
});

export const updateGameMode = mode => {
  return () => {
    socket.emit(UPDATE_GAME_MODE, mode);
  };
};
