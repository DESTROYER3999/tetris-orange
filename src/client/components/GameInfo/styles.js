import styled from "styled-components";
import { withStyles } from "@material-ui/core/styles";
import MenuItem from "@material-ui/core/MenuItem";

export const StyledContainer = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  // justify-content: space-around;
  padding: 20px;
  margin: 5px;
  background-color: rgba(0, 0, 0, 0.3);
  border-radius: 2px 60px 2px 2px;
  .player-count {
    font-size: 2vh;
    flex: 1;
    display: flex;
    align-items: flex-end;
  }
`;

export const StyledRowContainer = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
  padding: 10px;
  .description {
    display: flex;
    flex: 1;
    flex-direction: column;
    align-items: flex-end;
  }
`;

export const StyledLabel = styled.label`
  font-size: 2vh;
  padding: 0.5vh;
`;

export const StyledMenuItem = withStyles({})(MenuItem);

export const StyledDescription = styled.p`
  font-family: Verdana;
  font-size: 1.1vh;
`;

export const StyledRoomName = styled.div`
  font-size: 3vh;
`;
