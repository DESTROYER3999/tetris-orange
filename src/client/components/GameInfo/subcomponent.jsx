import React from "react";
import PropTypes from "prop-types";
import {
  StyledContainer,
  StyledLabel,
  StyledMenuItem,
  StyledRowContainer,
  StyledDescription,
  StyledRoomName
} from "./styles";

import Select from "../common/Select";
import {
  SOLO,
  BATTLEROYAL,
  MAX_PLAYER_SOLO,
  MAX_PLAYER_BATTLEROYAL
} from "../../../constants/game";

const GameInfoSub = ({
  mode,
  playersCount,
  name,
  isStarted,
  isHost,
  onChangeGameMode
}) => {
  const maxPlayer = mode === SOLO ? MAX_PLAYER_SOLO : MAX_PLAYER_BATTLEROYAL;
  const handleModeChange = e => {
    onChangeGameMode(e.target.value);
  };

  const displayModeDescription = () => {
    if (mode === SOLO) {
      return (
        <StyledDescription>
          Single player mode, try to reach the highest score !
        </StyledDescription>
      );
    } else {
      return (
        <StyledDescription>
          Multiplayer mode, be the last one alive to win ! Removing multiple
          lines at a time send malus blocks to your opponents.
        </StyledDescription>
      );
    }
  };

  return (
    <StyledContainer id="game-info">
      <StyledRoomName>{name}</StyledRoomName>
      <StyledRowContainer>
        <Select
          disabled={!isHost || isStarted}
          value={mode}
          onChange={handleModeChange}
          label="Mode"
        >
          <StyledMenuItem disabled={playersCount > 1} value={SOLO}>
            Solo
          </StyledMenuItem>
          <StyledMenuItem value={BATTLEROYAL}>Battleroyal</StyledMenuItem>
        </Select>
        <div className="description">
          <StyledLabel>Description</StyledLabel>
          {displayModeDescription()}
        </div>
      </StyledRowContainer>
      <div className="player-count">
        [{playersCount}/{maxPlayer} Players]
      </div>
    </StyledContainer>
  );
};

GameInfoSub.propTypes = {
  mode: PropTypes.string.isRequired,
  playersCount: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  isStarted: PropTypes.bool.isRequired,
  isHost: PropTypes.bool.isRequired,
  onChangeGameMode: PropTypes.func.isRequired
};

export default GameInfoSub;
