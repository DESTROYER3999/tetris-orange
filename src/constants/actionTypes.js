export const JOIN_ROOM = "JOIN_ROOM";
export const UPDATE_ROOM = "UPDATE_ROOM";
export const LEAVE_ROOM = "LEAVE_ROOM";
export const LOBBY_ROOM = "LOBBY_ROOM";

export const NEW_PLAYER = "NEW_PLAYER";
export const DISCONNECT_PLAYER = "DISCONNECT_PLAYER";
export const UPDATE_PLAYER = "UPDATE_PLAYER";
export const UPDATE_PLAYERS_LIST = "UPDATE_PLAYERS_LIST";
export const DELETE_PLAYER = "DELETE_PLAYER";

export const HANDLE_HASH = "HANDLE_HASH";

export const UPDATE_BOARD = "UPDATE_BOARD";
export const UPDATE_GAME = "UPDATE_GAME";
export const UPDATE_GAME_MODE = "UPDATE_GAME_MODE";
export const UPDATE_SPECTRES = "UPDATE_SPECTRES";

export const DISPLAY_TOAST = "DISPLAY_TOAST";

export const CONNECTION = "connection";
export const DISCONNECT = "disconnect";

export const ADD_CHAT_MESSAGE = "ADD_CHAT_MESSAGE";
export const NEW_CHAT_MESSAGE = "NEW_CHAT_MESSAGE";

export const KEY_PRESSED = "KEY_PRESSED";
