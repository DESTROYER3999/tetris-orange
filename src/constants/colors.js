export const MALUS = "#3d3d3d";

export const VIVID_BLUE = "#143cdc";

export const CYAN_BLUE = "#14d4dc";

export const CRIMSON = "#dc143c";

export const PURE_ORANGE = "#ff8b00";

export const LIME_GREEN = "#3cdc14";

export const PURE_VIOLET = "#b600ff";

export const PURE_YELLOW = "#fff400";

export const DARK_CYAN = "#375f64";

export const GREEN_CHAT = "#73F054";

export const RED_CHAT = "#ff3366";
