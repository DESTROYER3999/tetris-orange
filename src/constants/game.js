export const SOLO = "Solo";
export const BATTLEROYAL = "Battleroyal";
export const MAX_PLAYER_SOLO = 1;
export const MAX_PLAYER_BATTLEROYAL = 5;
export const BOARD_HEIGHT = 20;
export const BOARD_WIDTH = 10;
