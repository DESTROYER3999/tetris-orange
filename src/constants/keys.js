export const enumKeys = {
  ARROW_LEFT: "ArrowLeft",
  ARROW_RIGHT: "ArrowRight",
  ARROW_DOWN: "ArrowDown",
  ARROW_UP: "ArrowUp",
  SPACE: "Space",
  ENTER: "Enter"
};
