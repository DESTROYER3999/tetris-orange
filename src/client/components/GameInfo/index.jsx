import { connect } from "react-redux";
import GameInfoSub from "./subcomponent";
import { updateGameMode } from "../../actions/game";

const mapStateToProps = state => ({
  mode: state.room.mode,
  playersCount: state.room.playersCount,
  name: state.room.name,
  isStarted: state.room.isStarted,
  isHost: state.player.isHost
});

const mapDispatchToProps = dispatch => {
  return {
    onChangeGameMode: mode => {
      dispatch(updateGameMode(mode));
    }
  };
};

const GameInfo = connect(
  mapStateToProps,
  mapDispatchToProps
)(GameInfoSub);

export default GameInfo;
