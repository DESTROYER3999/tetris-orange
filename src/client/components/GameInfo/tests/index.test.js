import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import GameInfo from "../index";
import socket from "../../../services/socket-api";
import thunk from "redux-thunk";
import { SOLO } from "../../../../constants/game";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {
  room: {
    mode: SOLO,
    playersCount: 1,
    name: "Player1",
    isStarted: false
  },
  player: { isHost: true },
  onChangeGameMode: jest.fn()
};

const store = mockStore(initialState);

beforeEach(done => {
  // Setup
  if (socket.disconnected) {
    socket.connect();
  }
  socket.on("connect", () => {
    done();
  });
});

afterEach(done => {
  //   Cleanup;
  if (socket.connected) {
    socket.disconnect();
  }
  done();
});

describe("<GameInfo />", () => {
  test("GameInfo rendering with store without crashing", () => {
    const wrapper = shallow(<GameInfo store={store} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
