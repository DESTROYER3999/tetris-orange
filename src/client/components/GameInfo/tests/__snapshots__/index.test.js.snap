// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameInfo /> GameInfo rendering with store without crashing 1`] = `
<ContextProvider
  value={null}
>
  <GameInfoSub
    isHost={true}
    isStarted={false}
    mode="Solo"
    name="Player1"
    onChangeGameMode={[Function]}
    playersCount={1}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
